docs/04_protocol/02_transactions_protocol.md

```diff
@@ -327,7 +327,7 @@ To execute the transaction, a chain database session is started and a snapshot i
 
 ### 3.5.1. Apply Context
 
-To prepare for action execution, an apply context is created locally for each action. The apply context, as its name implies, contains references to the necessary resources to apply the action, such as an instance to the chain controller (see [Network Peer Protocol: 2.2. Chain Controller](03_network_peer_protocol.md#22-chain-controller)), the chain database where state is kept, the transaction context where the transaction is running, the actual action instance, the receiver account to whom the action is intended, etc.
+To prepare for action execution, an apply context is created locally for each action. The apply context, as its name implies, contains references to the necessary resources to apply the action, such as an instance to the chain controller (see [Network Peer Protocol: 2.2. Chain Controller](03_network_peer_protocol.md#22-chain-controller)), the chain database where state is kept, the transaction context where the transaction is running, the actual action instance, and the receiver account to whom the action is intended.
```

Comment:
maybe this?
To prepare for action execution, an apply context is created locally for each action. -> Each action execution needs  an `apply context` object created in the local scope of the action. The `apply context`...

Reply:
The apply context instance is actually created within the scope of the transaction context, not within the local scope of the action. I added the word "instance" after apply context to make it more explicit.

Reply:
yeah ok, but I am surprised you do not notice the passive language used here `To prepare for action execution, an apply context is created locally for each action.`
that's why I've proposed a different way of putting it in active voice.

Reply:
Yes I noticed that you tried to make it less passive voiced - thanks! 👍 - but this is one of those exceptions where the subject that performs the action is not clear or it is redundant (it is the EOSIO software that creates the apply context instance, but that would sound too repetitive throughout the document wouldn't it?). Therefore, passive voice is fine in this case according to the google style. Btw I think your example also uses passive voice with respect to the action context being created. Cheers

Reply:
The subject that performs the action is not clear because it is passive voice you are using.
BUT you can avoid it by using active voice, and one ways is to do it the way I put it. I left out who is creating the context and made the subject the "action execution" which "needs an apply context".
We can discuss this more in our next team meeting.